### PR TITLE
[BEAM-22] Add InProcessRegistrar

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace "<Jira issue #>" in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Adds @AutoService for the InProcessPipelineRunner and
InProcessPipelineOptions

### CODE REVIEW

File: sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessRegistrar.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessRegistrar.java]: Please add a test like DataflowPipelineRegistrarTest

**Comment by lukecwik** [comment]
R: @lukecwik

File: sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessRegistrar.java (hunk 1 above)
Comment (tgroh) [inline sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessRegistrar.java]: Done.

**Comment by lukecwik** [comment]
LGTM

**Pull request closed without merging**